shared/src/main/scala/mlscript/codegen/Codegen.scala

```diff
@@ -122,15 +131,32 @@ object SourceCode {
   })
   def fromStmts(stmts: Ls[JSStmt]): SourceCode = SourceCode.concat(stmts map { _.toSourceCode })
 
+  val ampersand: SourceCode = SourceCode(" & ")
   val space: SourceCode = SourceCode(" ")
   val semicolon: SourceCode = SourceCode(";")
+  val colon: SourceCode = SourceCode(": ")
+  val separator: SourceCode = SourceCode(" | ")
   val comma: SourceCode = SourceCode(",")
   val commaSpace: SourceCode = SourceCode(", ")
   val empty: SourceCode = SourceCode(Nil)
+  val openCurlyBrace: SourceCode = SourceCode("{")
+  val closeCurlyBrace: SourceCode = SourceCode("}")
+  val openAngleBracket: SourceCode = SourceCode("<")
+  val closeAngleBracket: SourceCode = SourceCode(">")
+  val fatArrow: SourceCode = SourceCode(" => ")
+  val equalSign: SourceCode = SourceCode(" = ")
 
   def concat(codes: Ls[SourceCode]): SourceCode =
     codes.foldLeft(SourceCode.empty) { _ + _ }
 
+  /**
+    * Source entries with curly braces
+    * 
+    * e.g. [1, 1] = {1, 1}
```

Comment:
I don't understand what `[1, 1] = {1, 1}` means. Are you using Haskell/JS syntax to document Scala parameters?

Reply:
Removed the confusing example with better explanation.